src/prop_current/fss_antenna_gainsV1_1.py

```diff
@@ -0,0 +1,138 @@
+'''
+Created on 6 juil. 2017
+
+@author: Red Technologies
+Gerard Garnier
+'''
+import vincenty
+import numpy as npy
+import math
+
+def GetFSSAntennaGain(pfl, cbsdLat, cbsdLng, cbsdHeight, fssLat, fssLng, fssHeight, fssPointingElevation, fssPointingAzimuth, w1=None, w2=None):
+    """
+    Returns the FSS Earth Station Antenna Gain (R2-SGN-21).
+    Inputs:
+        pfl                       Terrain profile line
+        cbsdLat, cbsdLng          lat/lon of cbsd (deg)
+        cbsdHeight                height of the CBSD, in meters (AGL)
+        fssLat, fssLng            lat/lon of fss (deg) 
+        fssHeight                 height of the FSS, in meters (AGL)
+        fssPointingElevation      the FSS earth station antenna elevation (deg : 0 - 359)        
+        fssPointingAzimuth        FSS earth station antenna azimuth (deg : 0 - 359)
+        w1, w2                    weights 
+    Output:
+        FSS Earth Station Antenna Gain
+    """
+    d, azstart, AZ = vincenty.dist_bear_vincenty(cbsdLat, cbsdLng, fssLat, fssLng)
+
+    theta = [0., 0.]
+    GetElevationAngle(pfl, fssHeight, cbsdHeight, theta)
+    
+            
+    # off-axis angle
+    ThetaAngle = (180 * (math.acos( (math.cos(theta[0]) * math.cos(math.radians(fssPointingElevation))\
+                                        * math.cos(math.radians(fssPointingAzimuth - AZ))) + (math.sin(theta[0]) * math.sin(math.radians(fssPointingElevation))) )))/ math.pi
+        
+    Ggso, GgsoP = GetGgsoAndGgsoP(ThetaAngle)
+    
+    # If the FSS earth station registration data includes values for w1 and w2, 
+    # SAS shall use these values. Otherwise SAS shall assume w1=0 and w2=1.
+    if w1 is None or w2 is None:
+        FSSGain = GgsoP
+    else:
+        FSSGain = w1*Ggso + w2*GgsoP
+    
+    return FSSGain
+
+def GetElevationAngle(pfl, fssHeight, cbsdHeight, theta):
+    
+    """
+    Returns elevation angles of the horizons from each terminal.
+    extract and rewrite of the ITM horizon calculation routine hzns()
+    Inputs:
+        pfl          Terrain profile line
+        fssHeight   height of the mobile, in meters
+        cbsdHeight   height of the base station, in meters
+    Output:
+        theta : Elevation angles
+                 - theta[0] = mobile horizon distance, in radians
```

Comment:
It seems to me that the the theta calculated here are slopes not angles in radians.